ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -446,18 +444,33 @@ addBlock cfg blk m = Model {
           (Proxy @(BlockProtocol blk))
           (selectView (configBlock cfg) . getHeader)
           (currentChain m)
-      . filter (extendsImmutableChain . fst)
-      $ candidates
+      $ consideredCandidates
 
-addBlocks :: LedgerSupportsProtocol blk
+    -- Sometimes blocks are added not in order and therefore we cannot just look
+    -- at the last added block and push it to the set of known valid blocks. The
+    -- easiest (and considering the lengths of the generated chains) way is to
```

Comment:
Maybe we can reword the sentence between parentheses (it's a bit long and breaks the continuity)